### PR TITLE
Register Form Layout (Closes #25)

### LINKED ISSUE #25: Create registration form
Should be similar to #5. Also, don't worry about giving it functionality. That'll be done in the registration page issue. This issue should mainly cover laying out the component, anywhere where you would usually put a function call (such as submitting the form) should just be replaced with a prop

### PR DESCRIPTION
**Pull request by leestanley**
Looks Like: https://i.imgur.com/Rq3IgoI.png

**Issues:**
- I made the "Major" field just a regular text input. I was originally going to make a dropdown of all the engineering majors, but then I realized other majors were going to join as well. I could add an autocomplete of all the majors at UCSD as a solution. 
- ~~I hard coded the "Years" options to graduating in 2020-2023. There might be a better solution to this, but it seemed the easiest at the time.~~

### CODE REVIEW

File: src/components/RegisterForm/style.less [hunk 1]
```diff
@@ -0,0 +1,76 @@
+@import '../../styles/fonts.less';
```
Comment (eminguyen) [inline src/components/RegisterForm/style.less]: two space indents

File: src/components/RegisterForm/style.less [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	background: @blue-2;
+	border-radius: 10px;
+	display: flex;
+	font-family: 'Roboto';
+	height: @register-height;
+	justify-content: center;
+	text-align: center;
+	width: @db-card-width;
+	label {
+		color: @white;
```
Comment (eminguyen) [inline src/components/RegisterForm/style.less]: can you just set the color of the register card to be white or does that not work? If not, we should probably add a reset.less file

File: src/components/RegisterForm/style.less [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		line-height: 30px;
+	}
+	.formitems {
+    margin-bottom: 0px;
+    margin-top: 10px;
+		text-align: left;
+	}
+	.majoryear {
+		display: flex;
+    text-align: left;
```
Comment (eminguyen) [inline src/components/RegisterForm/style.less]: indents

File: src/components/RegisterForm/style.less [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+		text-align: left;
+		&:hover {
+			opacity: 0.9;
+		}
+	}
+	.ant-form-item-label {
+		line-height: 30px;
+	}
+	.formitems {
+    margin-bottom: 0px;
```
Comment (eminguyen) [inline src/components/RegisterForm/style.less]: indents

File: src/components/RegisterForm/style.less [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+		width: @button-width;
+		.major {
+			width: 170px;
+		}
+		.year {
+			width: 80px;
+		}
+	}
+	.names {
+		.majoryear;
```
Comment (eminguyen) [inline src/components/RegisterForm/style.less]: why

File: src/components/RegisterForm/index.jsx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+                name="major"
+                className="major"
+                value={props.values}
+                onChange={props.handleChange}
+                onBlur={props.handleBlur}
+              />
+            </Form.Item>
+            <Form.Item label="Year" className="formitems">
+              <Select className="year" value={props.values}>
+                <Option value="2020">2020</Option>
```
Comment (eminguyen) [inline src/components/RegisterForm/index.jsx]: can you make this an array at the top and then utilize a for loop to iterate through it? By doing it that way, it'll be convenient to add more values when we have to later on.

File: src/components/RegisterForm/index.jsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+              Already have an account? <b>Sign in.</b>
+            </p>
+          </NavLink>
+        </form>
+      </div>
+    </div>
+  );
+};
+
+// TODO: swap out proptypes with formik's implementation
```
Comment (eminguyen) [inline src/components/RegisterForm/index.jsx]: Capitalize Swap and add a period to the end

**Comment by sumeet-bansal** [comment]
@leestanley If you’re going to hardcode graduation year, do it for current year through current year plus six. How would autocomplete for majors work? Would you be hardcoding that or pulling the majors from somewhere dynamically?

**Comment by leestanley** [comment]
> @leestanley If you’re going to hardcode graduation year, do it for current year through current year plus six. How would autocomplete for majors work? Would you be hardcoding that or pulling the majors from somewhere dynamically?

For Ant Design's autocomplete implementation, you would need to pass in an array of all the majors. I would also need hardcode list of majors manually.

File: src/components/RegisterForm/style.less (hunk 2 above)
Comment (leestanley) [reply src/components/RegisterForm/style.less]: I tried targeting the register card but it didn't work

**Comment by sumeet-bansal** [comment]
> > @leestanley If you’re going to hardcode graduation year, do it for current year through current year plus six. How would autocomplete for majors work? Would you be hardcoding that or pulling the majors from somewhere dynamically?
> 
> For Ant Design's autocomplete implementation, you would need to pass in an array of all the majors. I would also need hardcode list of majors manually.

Can we have an open issue for pulling majors from some UCSD source then? Can you also generate the list of graduation years dynamically instead of hardcoding them? Otherwise, we'll need to update the array yearly.

File: src/components/RegisterForm/index.jsx [hunk 8]
```diff
@@ -7,9 +7,19 @@ import './style.less';
...
+  let currentYear = new Date().getFullYear();
+  let years = [];
+  for (var i = currentYear; i < currentYear + 6; i++) {
+    years.push(i.toString());
+  }
+  return years.concat(['Other']);
+}
+
+const years = getYears();
+
```
Comment (leestanley) [inline src/components/RegisterForm/index.jsx]: Here was my solution to the dynamic year update

File: src/components/RegisterForm/index.jsx [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+import PropTypes from 'prop-types';
+import { Form, Input, Button, Select } from 'antd';
+import { NavLink } from 'react-router-dom';
+
+import './style.less';
+import logo from '../../assets/graphics/logo.svg';
+
+const { Option } = Select;
+
+function getYears() {
```
Comment (leestanley) [inline src/components/RegisterForm/index.jsx]: Added the dynamic years

File: src/components/RegisterForm/index.jsx [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+  let currentYear = new Date().getFullYear();
+  let years = [];
+  for (var i = currentYear; i < currentYear + 6; i++) {
+    years.push(i.toString());
+  }
+  return years.concat(['Other']);
```
Comment (leestanley) [inline src/components/RegisterForm/index.jsx]: Added the dynamic years

Comment (sumeet-bansal) [reply src/components/RegisterForm/index.jsx]: Needs to be an int (not a string) within 6 six years of the current year (see schema [here](https://github.com/acmucsd/membership-portal/blob/master/app/db/models/user.js#L106) and example User instances [here](https://github.com/acmucsd/membership-portal/blob/master/app/db/populateDb.js)).

Also, the actual function could be a little tighter, e.g.
```javascript
function getYears() {
    const currentYear = new Date().getFullYear();
    return [...Array(6)].map((_,i) => i + currentYear);
}
```

**Pull request merged**